Bottom: 2d8664ed48f538eabfc7d3b0ec5bb5e365021672
Top:    cf428d606ffdc12333adef1792c7b450a68d8797
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-12 01:22:10 +0100

Debug / attempt to fix DRC crash


---

diff --git a/src/find.c b/src/find.c
index a19b23d..5d5687c 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2542,6 +2542,10 @@ IsLineInPolygon (LineType *Line, PolygonType *Polygon)
        Coord wid = (Line->Thickness + Bloat + 1) / 2;
        Coord x1, x2, y1, y2;
 
+       printf ("Line thickness is %i, Bloat is %i\n",
+               Line->Thickness, Bloat);
+       printf ("Testing with wid=%i\n", wid);
+
        x1 = MIN (Line->Point1.X, Line->Point2.X) - wid;
        y1 = MIN (Line->Point1.Y, Line->Point2.Y) - wid;
        x2 = MAX (Line->Point1.X, Line->Point2.X) + wid;
diff --git a/src/polygon.c b/src/polygon.c
index 74a7b7f..be7de17 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1694,6 +1694,7 @@ bool
 IsRectangleInPolygon (Coord X1, Coord Y1, Coord X2, Coord Y2, PolygonType *p)
 {
   POLYAREA *s;
+  printf ("X1=%i, X2=%i, Y1=%i, Y2=%i\n", X1, X2, Y1, Y2);
   if (!
       (s = RectPoly (min (X1, X2), max (X1, X2), min (Y1, Y2), max (Y1, Y2))))
     return false;
